api/proto/teleport/workloadidentity/v1/revocation_resource.proto

```diff
@@ -0,0 +1,54 @@
+// Copyright 2025 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package teleport.workloadidentity.v1;
+
+import "google/protobuf/timestamp.proto";
+import "teleport/header/v1/metadata.proto";
+
+option go_package = "github.com/gravitational/teleport/api/gen/proto/go/teleport/workloadidentity/v1;workloadidentityv1";
+
+// WorkloadIdentityX509Revocation represents the revocation of a single X509
+// workload identity credential. Creating or deleting these resources triggers
+// the regeneration of the trust domain CRL.
+//
+// The name of a WorkloadIdentityX509Revocation must be the base16, lower case,
+// encoded serial number of the revoked X509 certificate. Therefore, only a
+// single revocation entry can exist for a given certificate.
+message WorkloadIdentityX509Revocation {
+  // The kind of resource represented. For WorkloadIdentityX509Revocation
+  // resources, this is always `workload_identity_x509_revocation`.
+  string kind = 1;
+  // Differentiates variations of the same kind. All resources should
+  // contain one, even if it is never populated.
+  string sub_kind = 2;
```

Comment:
For my own learning, what is the purpose of `kind` here? If it's always `workload_identity_x509_revocation` why is it a member of this message rather than the containing message?

Reply:
It's a little awkward since we define this in all resources, although, we always expect it to be a fixed value for a specific resource. However, it does mean that if we marshal the resources to YAML or JSON, we can determine the type of an unknown resource when we want to unmarshal it. I suppose an alternative would've been some kind of custom marshal/unmarshaler that automatically appends this field with a fixed value.